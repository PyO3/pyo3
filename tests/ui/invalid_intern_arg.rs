use pyo3::Python;

fn main() {
    let _foo = if true { "foo" } else { "bar" };
    Python::with_gil(|py| py.import(pyo3::intern!(py, _foo)).unwrap());
}
