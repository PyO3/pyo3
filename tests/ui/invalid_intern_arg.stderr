error[E0435]: attempt to use a non-constant value in a constant
 --> tests/ui/invalid_intern_arg.rs:5:55
  |
5 |     Python::with_gil(|py| py.import(pyo3::intern!(py, _foo)).unwrap());
  |                                                       ^^^^ non-constant value
  |
help: consider using `let` instead of `static`
 --> src/sync.rs
  |
  -         static INTERNED: $crate::sync::Interned = $crate::sync::Interned::new($text);
  +         let INTERNED: $crate::sync::Interned = $crate::sync::Interned::new($text);
  |

error: lifetime may not live long enough
 --> tests/ui/invalid_intern_arg.rs:5:27
  |
5 |     Python::with_gil(|py| py.import(pyo3::intern!(py, _foo)).unwrap());
  |                       --- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'2`
  |                       | |
  |                       | return type of closure is pyo3::Bound<'2, PyModule>
  |                       has type `Python<'1>`
